testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestTablePartitioningWithSpecialChars.java

```diff
@@ -0,0 +1,118 @@
+            "(12, 'with space')";
+
+    private static final ImmutableList<QueryAssert.Row> EXPECTED_PARTITION_VALUES = ImmutableList.of(
+            row(1, "with-hyphen"),
+            row(2, "with.dot"),
+            row(3, "with:colon"),
+            row(4, "with/slash"),
+            row(5, "with\\\\backslashes"),
+            row(6, "with\\backslash"),
+            row(7, "with=equal"),
+            row(8, "with?question"),
+            row(9, "with!exclamation"),
+            row(10, "with%percent"),
+            row(11, "with%%percents"),
+            row(12, "with space"));
+
+    @Test(groups = HIVE_PARTITIONING)
+    public void testStringPartitioningWithSpecialCharactersCtasInTrino()
+    {
+        String tableName = "test_string_partitioning_with_special_chars_ctas_in_trino";
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + tableName);
+        onTrino().executeQuery(format("CREATE TABLE %s (id, part_col) " +
+                        "WITH (partitioned_by = ARRAY['part_col']) " +
+                        "AS VALUES " + INSERTED_PARTITION_VALUES,
+                tableName));
+
+        assertThat(onHive().executeQuery("SELECT * FROM " + tableName)).containsOnly(EXPECTED_PARTITION_VALUES);
+        assertThat(onTrino().executeQuery("SELECT * FROM " + tableName)).contains(EXPECTED_PARTITION_VALUES);
+    }
+
+    @Test(groups = HIVE_PARTITIONING)
+    public void testStringPartitioningWithSpecialCharactersInsertInTrino()
+    {
+        String tableName = "test_string_partitioning_with_special_chars_insert_in_trino";
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + tableName);
+        onTrino().executeQuery(format("CREATE TABLE %s (id BIGINT, part_col VARCHAR) WITH (partitioned_by = ARRAY['part_col']) ", tableName));
+        onTrino().executeQuery(format("INSERT INTO %s VALUES " + INSERTED_PARTITION_VALUES, tableName));
+
+        assertThat(onHive().executeQuery("SELECT * FROM " + tableName)).containsOnly(EXPECTED_PARTITION_VALUES);
+        assertThat(onTrino().executeQuery("SELECT * FROM " + tableName)).contains(EXPECTED_PARTITION_VALUES);
+    }
+
+    @Test(groups = HIVE_PARTITIONING)
+    public void testStringPartitioningWithSpecialCharactersInsertInHive()
+    {
+        String sourceTableName = "test_string_partitioning_with_special_chars_insert_in_hive_source";
+        String tableName = "test_string_partitioning_with_special_chars_insert_in_hive";
+
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + sourceTableName);
+        onTrino().executeQuery(format("CREATE TABLE %s (id, part_col) AS VALUES " + INSERTED_PARTITION_VALUES, sourceTableName));
+
+        onHive().executeQuery("DROP TABLE IF EXISTS " + tableName);
+        onHive().executeQuery(format("CREATE TABLE %s (id BIGINT) PARTITIONED BY (part_col STRING) ", tableName));
+        onHive().executeQuery("set hive.exec.dynamic.partition.mode=nonstrict"); // needed for dynamic partitioning with no static partitions in insert statement
```

Comment:
Does this mean some Hive distributions only support INSERT if the value of the partition column is a constant (within that statement)?

Reply:
It'd be nice to mention here which distributions required this change.

Reply:
I think we can leave this out since this is related to old test setup which doesn't exist anymore.